webdriver/client.py

```diff
@@ -385,12 +377,13 @@ def send_command(self, method, uri, body=None):
 
         url = urlparse.urljoin("session/%s/" % self.session_id, uri)
         response = self.send_raw_command(method, url, body)
+        value = response.body["value"]
 
-        rv = response.body["value"]
-        if not rv:
-            rv = None
+        if response.status != 200:
```

Comment:
It's looking like the best way forward is to simply recommend the usage of
`session.transport.send` in tests for specific commands. It will make the tests
a little more verbose (they'll have to inject the session ID themselves), but
this explicitness matches the spec (which consistently documents the session
ID's presence in the URI template). And we can always define a Pytest fixture to
abstract that chore, though I am always struggling to know the "correct" balance
between explicitness and conciseness.

In `master` right now, the caller determines when session ID takes place by
choosing between `send_command` and `send_raw_command`:

    send_command()         -> send_raw_command() -> transport.send()
    (inject session ID)       (raise exceptions)

Your most recent proposal has the caller using a utility function instead,
something like this:

    _session_url()
    (inject session ID)
          |
          V
    send_command()         -> transport.send()
    (raise exceptions)

And @jgraham has suggested the following (essentially equivalent to what we
have now, just with more accurate names)

    send_session_command() -> send_command()     -> transport.send()
    (inject session ID)       (raise exceptions)

I'd be satisfied with any, given that they all allow for using `transport.send`
directly in the cases where it seems appropriate.